[BUG]"io/ioutil" has been deprecated since Go 1.16

**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

Build failed due to static check.

![image](https://user-images.githubusercontent.com/3724388/211123986-87d1b8cf-0659-4f8b-aa4c-be7dcab8c4a1.png)


pkg/controllers/v1alpha1/databackup/implement.go:23:2: "io/ioutil" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

--- a/pkg/controllers/v1alpha1/databackup/implement.go
+++ b/pkg/controllers/v1alpha1/databackup/implement.go
@@ -20,7 +20,6 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"io/ioutil"
 	"os"
 	"reflect"
 	"strconv"
@@ -491,11 +490,11 @@ func (r *DataBackupReconcilerImplement) generateDataBackupValueFile(ctx reconcil
 		return
 	}
 
-	valueFile, err := ioutil.TempFile(os.TempDir(), fmt.Sprintf("%s-%s-%s-backuper-values.yaml", databackup.Namespace, databackup.Name, dataBackup.RuntimeType))
+	valueFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("%s-%s-%s-backuper-values.yaml", databackup.Namespace, databackup.Name, dataBackup.RuntimeType))
 	if err != nil {
 		return
 	}
-	err = ioutil.WriteFile(valueFile.Name(), data, 0400)
+	err = os.WriteFile(valueFile.Name(), data, 0400)
 	if err != nil {
 		return
 	}

--- a/pkg/csi/recover/recover.go
+++ b/pkg/csi/recover/recover.go
@@ -19,6 +19,11 @@ package recover
 import (
 	"context"
 	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
 	"github.com/fluid-cloudnative/fluid/pkg/common"
 	"github.com/fluid-cloudnative/fluid/pkg/utils"
 	"github.com/fluid-cloudnative/fluid/pkg/utils/dataset/volume"
@@ -26,7 +31,6 @@ import (
 	"github.com/fluid-cloudnative/fluid/pkg/utils/mountinfo"
 	"github.com/golang/glog"
 	"github.com/pkg/errors"
-	"io/ioutil"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
@@ -35,12 +39,8 @@ import (
 	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
 	k8sexec "k8s.io/utils/exec"
 	"k8s.io/utils/mount"
-	"os"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
-	"strconv"
-	"strings"
-	"time"
 )
 
 const (
@@ -74,7 +74,7 @@ type containerStat struct {
 
 func initializeKubeletClient() (*kubelet.KubeletClient, error) {
 	// get CSI sa token
-	tokenByte, err := ioutil.ReadFile(serviceAccountTokenFile)
+	tokenByte, err := os.ReadFile(serviceAccountTokenFile)
 	if err != nil {
 		return nil, errors.Wrap(err, "in cluster mode, find token failed")
 	}

--- a/pkg/csi/recover/recover_test.go
+++ b/pkg/csi/recover/recover_test.go
@@ -18,7 +18,7 @@ package recover
 
 import (
 	"errors"
-	"io/ioutil"
+	"os"
 	"reflect"
 	"testing"
 	"time"
@@ -77,7 +77,7 @@ func Test_initializeKubeletClient(t *testing.T) {
 				fakeClientKey      = ""
 				fakeKubeletTimeout = "120"
 			)
-			patch1 := ApplyFunc(ioutil.ReadFile, func(filename string) ([]byte, error) {
+			patch1 := ApplyFunc(os.ReadFile, func(filename string) ([]byte, error) {
 				return []byte(fakeToken), nil
 			})
 			defer patch1.Reset()

--- a/pkg/ddc/alluxio/load_data.go
+++ b/pkg/ddc/alluxio/load_data.go
@@ -18,7 +18,6 @@ package alluxio
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"strings"
 
@@ -128,11 +127,11 @@ func (e *AlluxioEngine) generateDataLoadValueFile(r cruntime.ReconcileRequestCon
 		return
 	}
 
-	valueFile, err := ioutil.TempFile(os.TempDir(), fmt.Sprintf("%s-%s-loader-values.yaml", dataload.Namespace, dataload.Name))
+	valueFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("%s-%s-loader-values.yaml", dataload.Namespace, dataload.Name))
 	if err != nil {
 		return
 	}
-	err = ioutil.WriteFile(valueFile.Name(), data, 0400)
+	err = os.WriteFile(valueFile.Name(), data, 0400)
 	if err != nil {
 		return
 	}

--- a/pkg/ddc/alluxio/master_internal.go
+++ b/pkg/ddc/alluxio/master_internal.go
@@ -17,7 +17,6 @@ package alluxio
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 
 	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
@@ -85,7 +84,7 @@ func (e *AlluxioEngine) generateAlluxioValueFile(runtime *datav1alpha1.AlluxioRu
 	}
 
 	//2. Get the template value file
-	valueFile, err := ioutil.TempFile(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", e.name, e.runtimeType))
+	valueFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", e.name, e.runtimeType))
 	if err != nil {
 		e.Log.Error(err, "failed to create value file", "valueFile", valueFile.Name())
 		return valueFileName, err
@@ -94,7 +93,7 @@ func (e *AlluxioEngine) generateAlluxioValueFile(runtime *datav1alpha1.AlluxioRu
 	valueFileName = valueFile.Name()
 	e.Log.V(1).Info("Save the values file", "valueFile", valueFileName)
 
-	err = ioutil.WriteFile(valueFileName, data, 0400)
+	err = os.WriteFile(valueFileName, data, 0400)
 	if err != nil {
 		return
 	}

--- a/pkg/ddc/eac/master_internal.go
+++ b/pkg/ddc/eac/master_internal.go
@@ -18,14 +18,14 @@ package eac
 
 import (
 	"fmt"
+	"os"
+
 	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
 	"github.com/fluid-cloudnative/fluid/pkg/common"
 	"github.com/fluid-cloudnative/fluid/pkg/utils"
 	"github.com/fluid-cloudnative/fluid/pkg/utils/helm"
 	"github.com/fluid-cloudnative/fluid/pkg/utils/kubeclient"
 	"github.com/fluid-cloudnative/fluid/pkg/utils/kubectl"
-	"io/ioutil"
-	"os"
 
 	"gopkg.in/yaml.v2"
 )
@@ -83,7 +83,7 @@ func (e *EACEngine) generateEACValueFile(runtime *datav1alpha1.EACRuntime) (valu
 	}
 
 	//2. Get the template value file
-	valueFile, err := ioutil.TempFile(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", e.name, e.runtimeType))
+	valueFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", e.name, e.runtimeType))
 	if err != nil {
 		e.Log.Error(err, "failed to create value file", "valueFile", valueFile.Name())
 		return valueFileName, err
@@ -92,7 +92,7 @@ func (e *EACEngine) generateEACValueFile(runtime *datav1alpha1.EACRuntime) (valu
 	valueFileName = valueFile.Name()
 	e.Log.V(1).Info("Save the values file", "valueFile", valueFileName)
 
-	err = ioutil.WriteFile(valueFileName, data, 0400)
+	err = os.WriteFile(valueFileName, data, 0400)
 	if err != nil {
 		return
 	}

--- a/pkg/ddc/goosefs/load_data.go
+++ b/pkg/ddc/goosefs/load_data.go
@@ -14,7 +14,6 @@ package goosefs
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"strings"
 
@@ -123,11 +122,11 @@ func (e *GooseFSEngine) generateDataLoadValueFile(r cruntime.ReconcileRequestCon
 		return
 	}
 
-	valueFile, err := ioutil.TempFile(os.TempDir(), fmt.Sprintf("%s-%s-loader-values.yaml", dataload.Namespace, dataload.Name))
+	valueFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("%s-%s-loader-values.yaml", dataload.Namespace, dataload.Name))
 	if err != nil {
 		return
 	}
-	err = ioutil.WriteFile(valueFile.Name(), data, 0400)
+	err = os.WriteFile(valueFile.Name(), data, 0400)
 	if err != nil {
 		return
 	}

--- a/pkg/ddc/goosefs/master_internal.go
+++ b/pkg/ddc/goosefs/master_internal.go
@@ -18,7 +18,6 @@ package goosefs
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 
 	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
@@ -86,7 +85,7 @@ func (e *GooseFSEngine) generateGooseFSValueFile(runtime *datav1alpha1.GooseFSRu
 	}
 
 	//2. Get the template value file
-	valueFile, err := ioutil.TempFile(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", e.name, e.runtimeType))
+	valueFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", e.name, e.runtimeType))
 	if err != nil {
 		e.Log.Error(err, "failed to create value file", "valueFile", valueFile.Name())
 		return valueFileName, err
@@ -95,7 +94,7 @@ func (e *GooseFSEngine) generateGooseFSValueFile(runtime *datav1alpha1.GooseFSRu
 	valueFileName = valueFile.Name()
 	e.Log.V(1).Info("Save the values file", "valueFile", valueFileName)
 
-	err = ioutil.WriteFile(valueFileName, data, 0400)
+	err = os.WriteFile(valueFileName, data, 0400)
 	if err != nil {
 		return
 	}

--- a/pkg/ddc/jindo/load_data.go
+++ b/pkg/ddc/jindo/load_data.go
@@ -18,7 +18,6 @@ package jindo
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"strings"
 
@@ -153,11 +152,11 @@ func (e *JindoEngine) generateDataLoadValueFile(r cruntime.ReconcileRequestConte
 		return
 	}
 
-	valueFile, err := ioutil.TempFile(os.TempDir(), fmt.Sprintf("%s-%s-loader-values.yaml", dataload.Namespace, dataload.Name))
+	valueFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("%s-%s-loader-values.yaml", dataload.Namespace, dataload.Name))
 	if err != nil {
 		return
 	}
-	err = ioutil.WriteFile(valueFile.Name(), data, 0400)
+	err = os.WriteFile(valueFile.Name(), data, 0400)
 	if err != nil {
 		return
 	}

--- a/pkg/ddc/jindo/master_internal.go
+++ b/pkg/ddc/jindo/master_internal.go
@@ -18,7 +18,6 @@ package jindo
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 
 	"github.com/fluid-cloudnative/fluid/pkg/utils"
@@ -63,7 +62,7 @@ func (e *JindoEngine) generateJindoValueFile() (valueFileName string, err error)
 	if err != nil {
 		return
 	}
-	valueFile, err := ioutil.TempFile(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", e.name, e.runtimeType))
+	valueFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", e.name, e.runtimeType))
 	if err != nil {
 		e.Log.Error(err, "failed to create value file", "valueFile", valueFile.Name())
 		return valueFileName, err
@@ -71,7 +70,7 @@ func (e *JindoEngine) generateJindoValueFile() (valueFileName string, err error)
 	valueFileName = valueFile.Name()
 	e.Log.V(1).Info("Save the values file", "valueFile", valueFileName)
 
-	err = ioutil.WriteFile(valueFileName, data, 0400)
+	err = os.WriteFile(valueFileName, data, 0400)
 	if err != nil {
 		return
 	}

--- a/pkg/ddc/juicefs/data_load.go
+++ b/pkg/ddc/juicefs/data_load.go
@@ -14,7 +14,6 @@ package juicefs
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strings"
@@ -165,11 +164,11 @@ func (j *JuiceFSEngine) generateDataLoadValueFile(r cruntime.ReconcileRequestCon
 	}
 	j.Log.Info("dataload value", "value", string(data))
 
-	valueFile, err := ioutil.TempFile(os.TempDir(), fmt.Sprintf("%s-%s-loader-values.yaml", dataload.Namespace, dataload.Name))
+	valueFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("%s-%s-loader-values.yaml", dataload.Namespace, dataload.Name))
 	if err != nil {
 		return
 	}
-	err = ioutil.WriteFile(valueFile.Name(), data, 0400)
+	err = os.WriteFile(valueFile.Name(), data, 0400)
 	if err != nil {
 		return
 	}

--- a/pkg/ddc/juicefs/master_internal.go
+++ b/pkg/ddc/juicefs/master_internal.go
@@ -18,7 +18,6 @@ package juicefs
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 
 	"sigs.k8s.io/yaml"
@@ -86,7 +85,7 @@ func (j *JuiceFSEngine) generateJuicefsValueFile(runtime *datav1alpha1.JuiceFSRu
 	}
 
 	//2. Get the template value file
-	valueFile, err := ioutil.TempFile(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", j.name, j.runtimeType))
+	valueFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", j.name, j.runtimeType))
 	if err != nil {
 		j.Log.Error(err, "failed to create value file", "valueFile", valueFile.Name())
 		return valueFileName, err
@@ -95,7 +94,7 @@ func (j *JuiceFSEngine) generateJuicefsValueFile(runtime *datav1alpha1.JuiceFSRu
 	valueFileName = valueFile.Name()
 	j.Log.Info("Save the values file", "valueFile", valueFileName)
 
-	err = ioutil.WriteFile(valueFileName, data, 0400)
+	err = os.WriteFile(valueFileName, data, 0400)
 	if err != nil {
 		return
 	}

--- a/pkg/ddc/thin/master_internal.go
+++ b/pkg/ddc/thin/master_internal.go
@@ -18,7 +18,6 @@ package thin
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 
 	"k8s.io/apimachinery/pkg/api/errors"
@@ -89,7 +88,7 @@ func (t *ThinEngine) generateThinValueFile(runtime *datav1alpha1.ThinRuntime, pr
 	}
 
 	//2. Get the template value file
-	valueFile, err := ioutil.TempFile(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", t.name, t.runtimeType))
+	valueFile, err := os.CreateTemp(os.TempDir(), fmt.Sprintf("%s-%s-values.yaml", t.name, t.runtimeType))
 	if err != nil {
 		t.Log.Error(err, "failed to create value file", "valueFile", valueFile.Name())
 		return valueFileName, err
@@ -98,7 +97,7 @@ func (t *ThinEngine) generateThinValueFile(runtime *datav1alpha1.ThinRuntime, pr
 	valueFileName = valueFile.Name()
 	t.Log.Info("Save the values file", "valueFile", valueFileName)
 
-	err = ioutil.WriteFile(valueFileName, data, 0400)
+	err = os.WriteFile(valueFileName, data, 0400)
 	if err != nil {
 		return
 	}

--- a/pkg/dump/dump.go
+++ b/pkg/dump/dump.go
@@ -2,7 +2,6 @@ package dump
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"os/signal"
 	"runtime"
@@ -75,7 +74,7 @@ func InstallgoroutineDumpGenerator() {
 func coredump(fileName string) {
 	log.Info("Dump stacktrace to file", "fileName", fileName)
 	trace := StackTrace(true)
-	err := ioutil.WriteFile(fileName, []byte(trace), 0644)
+	err := os.WriteFile(fileName, []byte(trace), 0644)
 	if err != nil {
 		log.Error(err, "Failed to write coredump.")
 	}

--- a/pkg/utils/charts_test.go
+++ b/pkg/utils/charts_test.go
@@ -16,7 +16,6 @@ limitations under the License.
 package utils
 
 import (
-	"io/ioutil"
 	"os"
 	"testing"
 )
@@ -32,7 +31,7 @@ func TestPathExists(t *testing.T) {
 }
 
 func TestGetChartsDirectory(t *testing.T) {
-	f, err := ioutil.TempFile("", "test")
+	f, err := os.CreateTemp("", "test")
 	if err != nil {
 		t.Errorf("MkdirTemp failed due to %v", err)
 	}

--- a/pkg/utils/helm/helm.go
+++ b/pkg/utils/helm/helm.go
@@ -17,7 +17,6 @@ package helm
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"os/exec"
 	"path/filepath"
@@ -41,7 +40,7 @@ var helmCmd = []string{"ddc-helm"}
 // It returns the name of the value file and error
 func GenerateValueFile(values interface{}) (valueFileName string, err error) {
 	// 1. generate the template file
-	valueFile, err := ioutil.TempFile(os.TempDir(), "values")
+	valueFile, err := os.CreateTemp(os.TempDir(), "values")
 	if err != nil {
 		log.Error(err, "Failed to create tmp file", "tmpfile", valueFile.Name())
 		return "", err
@@ -60,7 +59,7 @@ func GenerateValueFile(values interface{}) (valueFileName string, err error) {
 // returns generated template file: templateFileName
 func GenerateHelmTemplate(name string, namespace string, valueFileName string, chartName string, options ...string) (templateFileName string, err error) {
 	tempName := fmt.Sprintf("%s.yaml", name)
-	templateFile, err := ioutil.TempFile("", tempName)
+	templateFile, err := os.CreateTemp("", tempName)
 	if err != nil {
 		return templateFileName, err
 	}

--- a/pkg/utils/kubectl/configmap.go
+++ b/pkg/utils/kubectl/configmap.go
@@ -17,7 +17,6 @@ package kubectl
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"os/exec"
 	"strings"
@@ -59,7 +58,7 @@ func SaveConfigMapToFile(name string, key string, namespace string) (fileName st
 		return "", err
 	}
 
-	file, err := ioutil.TempFile(os.TempDir(), name)
+	file, err := os.CreateTemp(os.TempDir(), name)
 	if err != nil {
 		log.Error(err, "failed to create tmp file", "tmpFile", file.Name())
 		return fileName, err

--- a/pkg/utils/mount.go
+++ b/pkg/utils/mount.go
@@ -18,19 +18,19 @@ package utils
 import (
 	"errors"
 	"fmt"
-	"github.com/golang/glog"
-	"io/ioutil"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/utils/mount"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"strings"
+
+	"github.com/golang/glog"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/mount"
 )
 
 const MountRoot string = "MOUNT_ROOT"
 
-//GetMountRoot gets the value of the env variable named MOUNT_ROOT
+// GetMountRoot gets the value of the env variable named MOUNT_ROOT
 func GetMountRoot() (string, error) {
 	mountRoot := os.Getenv(MountRoot)
 
@@ -60,7 +60,7 @@ func IsMounted(absPath string) (bool, error) {
 		return false, err
 	}
 
-	file, err := ioutil.ReadFile("/proc/mounts")
+	file, err := os.ReadFile("/proc/mounts")
 	if err != nil {
 		return false, err
 	}

--- a/pkg/utils/mount_test.go
+++ b/pkg/utils/mount_test.go
@@ -2,7 +2,6 @@ package utils
 
 import (
 	"errors"
-	"io/ioutil"
 	"os"
 	"os/exec"
 	"reflect"
@@ -96,7 +95,7 @@ func TestIsMounted(t *testing.T) {
 				return nil, nil
 			})
 			defer patch2.Reset()
-			patch1 := ApplyFunc(ioutil.ReadFile, func(filename string) ([]byte, error) {
+			patch1 := ApplyFunc(os.ReadFile, func(filename string) ([]byte, error) {
 				return []byte("JuiceFS:minio /var/lib/kubelet/pods/4781fc5b-72f9-4175-9321-2e1f169880ce/volumes/kubernetes.io~csi/default-jfsdemo/mount fuse.juicefs rw,relatime,user_id=0,group_id=0,default_permissions,allow_other 0 0"), nil
 			})
 			defer patch1.Reset()
@@ -107,7 +106,7 @@ func TestIsMounted(t *testing.T) {
 			So(mounted, ShouldBeTrue)
 		})
 		Convey("IsMounted false", func() {
-			patch1 := ApplyFunc(ioutil.ReadFile, func(filename string) ([]byte, error) {
+			patch1 := ApplyFunc(os.ReadFile, func(filename string) ([]byte, error) {
 				return []byte("JuiceFS:minio /var/lib/kubelet/pods/4781fc5b-72f9-4175-9321-2e1f169880ce/volumes/kubernetes.io~csi/default-jfsdemo/mount fuse.juicefs rw,relatime,user_id=0,group_id=0,default_permissions,allow_other 0 0"), nil
 			})
 			defer patch1.Reset()
@@ -122,7 +121,7 @@ func TestIsMounted(t *testing.T) {
 			So(mounted, ShouldBeFalse)
 		})
 		Convey("token len is 1", func() {
-			patch1 := ApplyFunc(ioutil.ReadFile, func(filename string) ([]byte, error) {
+			patch1 := ApplyFunc(os.ReadFile, func(filename string) ([]byte, error) {
 				return []byte("JuiceFS:minio "), nil
 			})
 			defer patch1.Reset()

--- a/pkg/utils/webhook/writer/fs.go
+++ b/pkg/utils/webhook/writer/fs.go
@@ -18,7 +18,6 @@ package writer
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"path"
 
@@ -41,31 +40,31 @@ func WriteCertsToDir(dir string, certs *generator.Artifacts) error {
 
 	// write or overwrite the cert for ca
 	filename := path.Join(dir, CAKeyName)
-	if err := ioutil.WriteFile(filename, certs.CAKey, 0600); err != nil {
+	if err := os.WriteFile(filename, certs.CAKey, 0600); err != nil {
 		return err
 	}
 	filename = path.Join(dir, CACertName)
-	if err := ioutil.WriteFile(filename, certs.CACert, 0600); err != nil {
+	if err := os.WriteFile(filename, certs.CACert, 0600); err != nil {
 		return err
 	}
 
 	// write or overwrite the certs for serverName
 	filename = path.Join(dir, ServerCertName)
-	if err := ioutil.WriteFile(filename, certs.Cert, 0600); err != nil {
+	if err := os.WriteFile(filename, certs.Cert, 0600); err != nil {
 		return err
 	}
 	filename = path.Join(dir, ServerKeyName)
-	if err := ioutil.WriteFile(filename, certs.Key, 0600); err != nil {
+	if err := os.WriteFile(filename, certs.Key, 0600); err != nil {
 		return err
 	}
 
 	// write or overwrite the certs for serverName2
 	filename = path.Join(dir, ServerCertName2)
-	if err := ioutil.WriteFile(filename, certs.Cert, 0600); err != nil {
+	if err := os.WriteFile(filename, certs.Cert, 0600); err != nil {
 		return err
 	}
 	filename = path.Join(dir, ServerKeyName2)
-	if err := ioutil.WriteFile(filename, certs.Key, 0600); err != nil {
+	if err := os.WriteFile(filename, certs.Key, 0600); err != nil {
 		return err
 	}
 	return nil

--- a/pkg/utils/yaml_test.go
+++ b/pkg/utils/yaml_test.go
@@ -1,13 +1,13 @@
 package utils
 
 import (
-	"gopkg.in/yaml.v2"
-	"io/ioutil"
 	"math/rand"
 	"os"
 	"reflect"
 	"strconv"
 	"testing"
+
+	"gopkg.in/yaml.v2"
 )
 
 type dummyOuter struct {
@@ -22,7 +22,7 @@ type dummyInner struct {
 }
 
 func TestToYaml(t *testing.T) {
-	tempFile, err := ioutil.TempFile(os.TempDir(), "dummy")
+	tempFile, err := os.CreateTemp(os.TempDir(), "dummy")
 	if err != nil {
 		t.Errorf("TestToYaml can't write temp file. error = %v", err)
 		t.FailNow()
@@ -47,9 +47,9 @@ func TestToYaml(t *testing.T) {
 	}
 
 	tempFileName := tempFile.Name()
-	bytes, err := ioutil.ReadFile(tempFileName)
+	bytes, err := os.ReadFile(tempFileName)
 	if err != nil {
-		t.Errorf("ioutil.ReadFile() error = %v, expected error = nil", err)
+		t.Errorf("os. ReadFile() error = %v, expected error = nil", err)
 		t.FailNow()
 	}
 

--- a/pkg/webhook/certificate_builder_test.go
+++ b/pkg/webhook/certificate_builder_test.go
@@ -18,7 +18,6 @@ package webhook
 
 import (
 	"context"
-	"io/ioutil"
 	"os"
 	"reflect"
 	"testing"
@@ -101,7 +100,7 @@ func TestBuildOrSyncCABundle(t *testing.T) {
 			t.Setenv(common.MyPodNamespace, item.ns)
 		}
 
-		certDir, err := ioutil.TempDir("/tmp", item.certPath)
+		certDir, err := os.MkdirTemp("/tmp", item.certPath)
 		if err != nil {
 			t.Errorf("MkdirTemp failed due to %v", err)
 		}

--- a/test/e2etest/patch_node_label_test.go
+++ b/test/e2etest/patch_node_label_test.go
@@ -3,19 +3,20 @@ package e2etest
 import (
 	"context"
 	"fmt"
+	"math/rand"
+	"os"
+	"strconv"
+	"time"
+
 	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	"io/ioutil"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
-	"math/rand"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/yaml"
-	"strconv"
-	"time"
 )
 
 var _ = Describe("Test alluxioruntime controller patch node labels", func() {
@@ -356,7 +357,7 @@ var _ = Describe("Test alluxioruntime controller patch node labels", func() {
 
 // readFile will read a yaml k8s object to runtime.Object.
 func readFile(fileName string, object runtime.Object) error {
-	data, err := ioutil.ReadFile(fileName)
+	data, err := os.ReadFile(fileName)
 	if err != nil {
 		return err
 	}

